### DISCUSSION TITLE
notprivacysafe.graphql.execution.SimpleDataFetcherExceptionHandler 47 [armeria-eventloop-epoll-4-1] WARN  [] - Exception while fetching data (/topology) : Service id is empty

### DISCUSSION THREAD

**Discussion by DoNotBugPlz** [apache/skywalking#10652] [Q&A]
env:  kubernetes/kubesphere deployment
skywalking-oap/ui/agent version:  9.4.0/9.4.0/8.14.0-java8
description: 
java application is standby like this:
<img width="1681" alt="image" src="https://user-images.githubusercontent.com/42760659/230301923-ddf8edc5-b667-4b34-9a6d-93cda0dc9f87.png">
skywalking-oap & skywalking-ui  is both working 
<img width="1153" alt="image" src="https://user-images.githubusercontent.com/42760659/230302947-aaec98c3-9441-4abb-9d9f-69daec1c0e29.png">
SomePodConfig post here:
OAP:
<img width="973" alt="image" src="https://user-images.githubusercontent.com/42760659/230303207-7187abd2-0c5d-4c66-ba71-ac7eb6d257c5.png">
UI:
<img width="971" alt="image" src="https://user-images.githubusercontent.com/42760659/230303333-bbd59a8a-e25b-406b-920e-f671a844634c.png">

But ,I haven't seen  my application on UI view.
check the skywalking-oap-server logs:
2023-04-06 07:02:31,112 notprivacysafe.graphql.execution.SimpleDataFetcherExceptionHandler 47 [armeria-eventloop-epoll-4-1] WARN  [] - Exception while fetching data (/topology) : Service id is empty

 org.apache.skywalking.oap.server.core.UnexpectedException: Service id is empty

<img width="1580" alt="image" src="https://user-images.githubusercontent.com/42760659/230304094-02b9d2e8-a800-4c57-9daf-7da3950fd1c5.png">

So how can I fix this problem ,many thanks.

**Comment by wu-sheng** [comment]
Try to debug how and which service have empty name. Maybe some exit spans cause that.
But we can't tell from your screenshots.

**Reply by DoNotBugPlz** [reply]
Thank you for your attention ，I just deploy only one application  which on k8s sidercar pattern，so I'd like to provider serial pods config , hopes to make my problem desp more  clearly :
![image](https://user-images.githubusercontent.com/42760659/230417567-9e5e5ccf-6e04-4d42-8a88-cd43b731894a.png)
this my application deployment:
![Snipaste_2023-04-06_22-59-02](https://user-images.githubusercontent.com/42760659/230419721-71b6c627-7b82-4399-a04a-2e29a0842efb.png)
![Snipaste_2023-04-06_22-59-37](https://user-images.githubusercontent.com/42760659/230419753-db906775-df41-449b-90dd-6abea3e01467.png)

**Reply by DoNotBugPlz** [reply]
Ok, Done! Port configuration not correct。It should be  11800
